spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/RewriteRowLevelOperationHelper.scala

```diff
@@ -240,6 +244,25 @@ trait RewriteRowLevelOperationHelper extends PredicateHelper with Logging {
     }
   }
 
+  private object Lit {
+    def unapply[T](literal: Literal[T]): Some[(T, DataType)] = {
+      Some((literal.value, literal.dataType))
+    }
+  }
+
+  private object TruncateTransform {
+    def unapply(transform: Transform): Option[(FieldReference, Int)] = transform match {
+      case at @ ApplyTransform(name, _) if name.equalsIgnoreCase("truncate")  =>
+        at.args match {
+        case Seq(nf: NamedReference, Lit(value: Int, IntegerType)) =>
+          Some(FieldReference(nf.fieldNames()), value)
```

Comment:
I think the convention in Spark is to have the transform width first, like `bucket(4, id)`. In Iceberg, we usually detect it in either location. Could you add a case for `Seq(Lit, NamedReference)` to handle that order as well?